build-deps.sh

```diff
@@ -101,33 +121,49 @@ function build() {
   cmake -S libgit2 -B libgit2/build\
     -DUSE_SSH=ON \
     -DBUILD_TESTS=OFF \
-    -DCMAKE_PREFIX_PATH="$deps;$deps/include/openssl" \
+    -DCMAKE_PREFIX_PATH="$deps" \
     -DCMAKE_INSTALL_PREFIX="$deps" \
     -DCMAKE_IGNORE_PREFIX_PATH="/usr" \
     -DCMAKE_OSX_ARCHITECTURES=$CMAKE_ARCH \
+    -DBUILD_SHARED_LIBS=OFF \
     -DCMAKE_BUILD_TYPE=Release
   cmake --build libgit2/build --target install
 
-  mkdir -p native/$OS_ARCH/
-  cp -vL $deps/{lib,lib64}/libcrypto.{dylib,so} native/$OS_ARCH/ || true
-  cp -vL $deps/{lib,lib64}/libssl.{dylib,so} native/$OS_ARCH/ || true
-  cp -vL $deps/{lib,lib64}/libssh2.{dylib,so} native/$OS_ARCH/ || true
-  cp -vL $deps/lib/libgit2.{dylib,so} native/$OS_ARCH/ || true
+  linkerOpts="$(PKG_CONFIG_PATH=`pwd`/deps/lib/pkgconfig pkg-config --libs libgit2 --static)"
+
+  mkdir -p src/nativeInterop/cinterop
+  cat > src/nativeInterop/cinterop/libgit2.def <<EOF
+package = com.github.git2
+headers = git2.h
+staticLibraries = libgit2.a
```

Comment:
This probably needs more entries. Sqlite, curl, OpenSSL, OpenSSL...

Reply:
I was assuming that this was the leaf and by including it then it would include all the other ones... I don't see how this would be working locally otherwise...

Reply:
That only accounts for SSH/SSL. Curl and sqlite aren't used by git.

Reply:
True.